celery/backends/database/__init__.py

```diff
@@ -84,57 +91,83 @@ def __init__(self, dburi=None, engine_options=None, url=None, **kwargs):
             conf.database_short_lived_sessions)
 
         tablenames = conf.database_table_names or {}
-        Task.__table__.name = tablenames.get('task', 'celery_taskmeta')
-        TaskSet.__table__.name = tablenames.get('group', 'celery_tasksetmeta')
+        self.task_cls.__table__.name = tablenames.get('task',
+                                                      'celery_taskmeta')
+        self.taskset_cls.__table__.name = tablenames.get('group',
+                                                         'celery_tasksetmeta')
 
         if not self.url:
             raise ImproperlyConfigured(
                 'Missing connection string! Do you have the'
                 ' database_url setting set to a real value?')
 
+    @property
+    def extended_result(self):
+        return self.app.conf.find_value_for_key('extended', 'result')
+
     def ResultSession(self, session_manager=SessionManager()):
         return session_manager.session_factory(
             dburi=self.url,
             short_lived_sessions=self.short_lived_sessions,
             **self.engine_options)
 
     @retry
-    def _store_result(self, task_id, result, state,
-                      traceback=None, max_retries=3, **kwargs):
+    def _store_result(self, task_id, result, state, traceback=None,
+                      request=None, **kwargs):
         """Store return value and state of an executed task."""
         session = self.ResultSession()
         with session_cleanup(session):
-            task = list(session.query(Task).filter(Task.task_id == task_id))
+            task = list(session.query(self.task_cls).filter(self.task_cls.task_id == task_id))
             task = task and task[0]
             if not task:
-                task = Task(task_id)
+                task = self.task_cls(task_id)
                 session.add(task)
                 session.flush()
-            task.result = result
-            task.status = state
-            task.traceback = traceback
+
+            self._update_result(task, result, state, traceback=traceback, request=request)
             session.commit()
-            return result
+
+    def _update_result(self, task, result, state, traceback=None,
+                       request=None):
+        task.result = result
+        task.status = state
+        task.traceback = traceback
+        if self.app.conf.find_value_for_key('extended', 'result'):
+            task.name = getattr(request, 'task_name', None)
```

Comment:
you follow a different coding pattern?

Reply:
No. @adimux  did the coding as I didn't have the time.